include/structures/junction.hpp

```diff
@@ -1,12 +1,18 @@
 #pragma once
 
+#include <seqan3/alphabet/nucleotide/dna5.hpp>
+#include <seqan3/range/views/char_to.hpp>
+#include <seqan3/range/views/complement.hpp>
+#include <seqan3/range/views/to.hpp>
+
 #include "structures/breakend.hpp"
 
 class Junction
 {
 private:
     Breakend mate1{};
     Breakend mate2{};
+    seqan3::dna5_vector inserted_sequence{};
```

Comment:
I think we should update the equality function so that a junction is only equal if its inserted_sequence is also equal (or has a certain minimum edit distance?).
EDIT: You decided the are equal and we save all different insertions in a vector in the cluster right? We should maybe add this as a comment to the equality function.

Reply:
Good point. I'll add the comparison of the inserted sequence to the comparison operators.
I'm not entirely sure I understand your second sentence. What do you suggest we add as a comment?

Reply:
Sorry for the confusion.
I just thought. We may want insertions with unequal sequences to be equal if the breakends are equal. Or maybe with similar sequences. But for now, maybe that's enough. But if you think we should add this later, you could add a TODO to the operator==() function.

Reply:
Thanks for the explanation, @Irallia. I now see what you mean and it is a very good question.
In my mind, each junction stands for an operation on the genome and the complete set of junctions should, in theory, transform the reference genome into the patient genome. Two junctions that have identical breakends but different insertion sequences are not identical because they yield different genome sequences when applied to the reference genome. Therefore, I would prefer the operator==() function to implement a strict identity check.

You are right that this affects our (simple) clustering method and leads to very small clusters because only identical junctions get clustered together. However, I think this is not the problem of the operator==() function (which I would expect to be strict). Instead, it is because the simple clustering method uses the operator==() function and we should soon implement a clustering method that makes more sense. Like you write, we would like to cluster junctions with similar breakends and similar insertion sequences (not only identical junctions). But in my opinion this should be done in a better clustering function, not in the operator==() function.